### DISCUSSION TITLE
Two way relation

### DISCUSSION THREAD

**Discussion by hakon-metzum** [pocketbase/pocketbase#124] [Q&A]
First of all - what a great project this is. I really love the simplicity, yet powerful options.

I have a question regarding relations. More specific - two way relations. Let me write an example: 

You have a project (project) with tasks (task). The task has a relation to the project (task->project). But is there a "simple" way to get all tasks for a project (project->tasks), without manually add the tasks as relations to the project?

Thanks!

**Comment by ganigeorgiev** [comment]
I think you could just filter by the task's project relation field. Assuming that your relation field is named `project`, here is an example using the JS-SDK:
```js 
const result = await client.Records.getList("tasks", 1, 50, {
     filter: "project = 'YOUR_PROJECT_ID' ",
})
```
You can even filter through the project fields if you want to match by name for example instead of id, and even combine with other fields from your project collection, eg:
`filter: "project.name = 'YOUR_PROJECT_NAME' && projet.status = 'active' "`

---

You can even define admin level filters, aka. filters that cannot be overwritten from the client-side and act as access rules. For example Let's say that your project collection has a `user` field and you want to show only the tasks that the current logged in user have access to. You can set the following simple API rule expression in your `tasks` collection settings (for each action - list, view, create, update, delete):
```
project.user = @request.user.id
```

**Reply by hakon-metzum** [reply]
Yeah. That could work. Thanks.
